### PR TITLE
Add Lix

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/SimonN/LixD/pull/437
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/SimonN/LixD/pull/437

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.lixgame.Lix

**Comment by Mailaender** [comment]
bot, build com.lixgame.Lix

File: com.lixgame.Lix.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.dmd"
+    ],
+    "command": "lix",
+    "finish-args": [
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline com.lixgame.Lix.json:12]:
```suggestion
        "--socket=fallback-x11",
        "--socket=wayland",
```

assuming it supports wayland

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.lixgame.Lix.json (hunk 1 above)
Comment (Mailaender) [reply com.lixgame.Lix.json]:
```
object.Exception@src/hardware/display.d(89): Can't instantiate a display even though we want to run Lix in interactive mode. This is very strange; normally, at least a window of size 640x480 should have spawned.
----------------
??:? pure @safe noreturn std.exception.bailOut!(Exception).bailOut(immutable(char)[], ulong, scope const(char)[]) [0x55d5453c3e1a]
??:? pure @safe allegro5.display.ALLEGRO_DISPLAY* std.exception.enforce!().enforce!(allegro5.display.ALLEGRO_DISPLAY*).enforce(allegro5.display.ALLEGRO_DISPLAY*, lazy const(char)[], immutable(char)[], ulong) [0x55d5453de7b4]
??:? void hardware.display.setScreenMode(in basics.cmdargs.Cmdargs) [0x55d54546b1da]
??:? void basics.resol.changeResolutionBasedOnCmdargsThenUserFile(const(basics.cmdargs.Cmdargs)) [0x55d54539d5bd]
??:? void basics.init.initialize(in basics.cmdargs.Cmdargs) [0x55d54539c5f6]
??:? int main.main(immutable(char)[][]).__lambda3() [0x55d54548cb40]
??:? extern (C) int allegro5.system.al_run_allegro(scope int delegate()).main_runner(int, char**) [0x55d5454d6f9d]
??:? int allegro5.system.al_run_allegro(scope int delegate()) [0x55d5454d6f7d]
??:? _Dmain [0x55d54548cb10]
```

Looks like Allegro 5 is not compatible with Wayland.

**Comment by Mailaender** [comment]
bot, build com.lixgame.Lix

File: com.lixgame.Lix.json [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.dmd"
+    ],
+    "command": "lix",
+    "finish-args": [
+        "--socket=x11",
+        "--share=ipc",
+        "--share=network",
+        "--device=dri"
```
Comment (barthalion) [inline com.lixgame.Lix.json:14]:
```suggestion
        "--device=dri",
        "--socket=pulseaudio"
```

**Comment by Mailaender** [comment]
bot, build com.lixgame.Lix

**Pull request closed without merging**